### PR TITLE
Feature/har converter correlate json

### PR DESCRIPTION
**Pull request by cyberw**
Add support for correlating json values in recordings, replacing recorded request values with references to the previous response.

Renamed includeCodeChecks flag to enableChecks (to match command line parameter name).

~~I'm having some issues with godep (I'm still a go noob...), so maybe the build will fail...~~ Actually I never added any dependencies so never mind that comment :)

### CODE REVIEW

**Comment by cyberw** [comment]
Hmm... Code coverage doesn't seem to be recorded for the integration tests (as evidenced by the failure of the last commit which failed because of behaviour on a line which is supposedly not covered :)

**Comment by cyberw** [comment]
@robingustafsson Any input? :)

**Comment by na--** [comment]
Since we already have a JavaScript runtime, I think it could be used for something like this in a very generic and flexible way (i.e. `k6 convert --processor cyberw.js my.har`). 

If we define some sort of API like we do for load tests, we can allow users to specify a modifier script with some exported functions in it. We can call those functions with the data/IR values/JSON objects/etc. before we actually write them to the output file so the script can modify them. Since we'll use a single runtime across all calls, the script can save some state between calls. Or if we have an IR, we can pass the whole file at once before we convert it to JS code.

File: converter/har/converter.go [hunk 1]
```diff
@@ -144,23 +180,42 @@ func Convert(h HAR, includeCodeCheck bool, returnOnFailedCheck bool, batchTime u
...
+							}
+						}
+					}
+
+					if e.Response.Headers != nil {
+						for _, header := range e.Response.Headers {
+							if header.Name == "Location" {
+								fmt.Fprintf(w, "\t\tredirectUrl = res.headers.Location;\n")
+								recordedRedirectURL = header.Value
+								guidRegexString := "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$"
```
[the remark this answers is not in the document]
Comment (cyberw) [reply converter/har/converter.go]: Agreed. The GUID stuff is very specific (although probably useful for a lot of rest services). I can leave out the GUID stuff if that helps with getting this merged :)

**Comment by cyberw** [comment]
I agree with @na-- , writing this stuff in go is messy to say the least. And using print-statements instead of some kind of AST is also quite crude... 

TBH I'm not sure this should even be a part of the same repo. sure the consistent command line parameter handling etc is good, but I think we're bloating the k6 a bit with stuff that is almost completely orthogonal to the runtime.

File: converter/har/converter.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (robingustafsson) [reply converter/har/converter.go]: Yes, I think we can merge this if the GUID stuff is removed.

**Comment by robingustafsson** [comment]
I agree that an IR combined with a JS based solution would be nice here (and maybe for other future uses as well as you point out @na--). Feels like a v2 of the HAR converter though 😄

File: converter/har/converter.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (cyberw) [reply converter/har/converter.go]: Done! I have another PR on the way (allowing the user to specify fields that should not be correlated). After that one I'd be happy to maybe have a chat about a more generic, pluggable solution, maybe along the lines of what @na-- suggested.

File: converter/har/converter.go [hunk 2]
```diff
@@ -97,16 +102,20 @@ func Convert(h HAR, includeCodeCheck bool, returnOnFailedCheck bool, batchTime u
 		sort.Sort(EntryByStarted(entries))
 
 		if nobatch {
-			var recordedRedirectUrl string
+			var recordedRedirectURL string
+			var recordedRestID string
```
Comment (na--) [inline converter/har/converter.go]: It seems like you've removed the setting of `recordedRedirectURL` and `recordedRestID` with the latest commit, but the variables themselves and checks with them that will never pass are still in the code

Comment (cyberw) [reply converter/har/converter.go]: good catch. I meant to remove recordedRestID, but recordedRedirectURL I meant to not remove the use of :) 

fixed now!

**Pull request merged**